docs/object-capabilities/derived-authority.md

```diff
@@ -0,0 +1,109 @@
+# Delegating and restricting authority
+
+> ## [Plan]
+>
+> * demonstrate basic use in combination with file system access etc.
+> * show how a derived authority can be constructed
+> * refer to the patterns for more complex scenarios
+>
+> ## [TODO]
+>
+> * [ ] come up with a better title for the section
+> * [ ] toc stuff and linking
+
+Any interesting program will need to interact with the outside world by using operating system concepts, like accessing the network or the file system, or by creating and communicating with other programs.
+The operating system has granted the right to do so to the program at startup (but it might do some checks at runtime, for example controlling access to a file). This is called __ambient authority__.
+
+The operating system has no concept of how our program is composed inside, and it knows nothing about Pony specific concepts in particular. Pony requires operations which interact with the system
+to be authorized, and allows us to restrict and delegate this authority inside our program.
+
+Recall the definition of capability from [ref]:
+
+> A capability is an unforgeable token that (a) designates an object and (b) gives the program the authority to perform a specific set of actions on that object.
+
+In Pony, the Main actor is passed the environment object, which keeps an `AmbientAuth` in the `root` field. This value is the capability that represents the ambient authority given to us by the system.
+Instead of allowing unfettered access to the outside world, Pony enforces the use of this authority.
+
+Here is a simple program that connects to example.com via TCP on port 80 and quits:
+
+```pony
+use "net"
+
+class MyTCPConnectionNotify is TCPConnectionNotify
+  let _out: OutStream
+
+  new iso create(out: OutStream) =>
+    _out = out
+
+  fun ref connected(conn: TCPConnection ref) =>
+    _out.print("connected")
+    conn.close()
+
+  fun ref connect_failed(conn: TCPConnection ref) =>
+    _out.print("connect_failed")
+
+actor Connect
+  new create(out: OutStream, auth: AmbientAuth) =>
+    TCPConnection(auth, MyTCPConnectionNotify(out), "example.com", "80")
+
+actor Main
+  new create(env: Env) =>
+    try Connect(env.out, env.root as AmbientAuth) end
+```
+
+The Main actor authorizes the Connect actor by passing the ambient authority token on to it. This token is not forgeable since the `AmbientAuth` constructor is private—only the runtime glue code
```

Comment:
Format `Main` and `Connect` please.

The only previous mention of token in this section is in the quote from the previous section. Can you work a mention of "ambient authority token" into this section earlier prior to the example?

Please use "unforgeable" rather than "not forgeable" so we state a consistent negation on forgeability.

Stating the runtime as "glue code" sounds dismissive to me. I understand what you mean though. Can you change it to something like: "This `AmbientAuthority` token is unforgeable since it is provided by the runtime itself so is guaranteed to have only one instance."

Reply:
```suggestion
The `Main` actor authorizes the `Connect` actor by passing the ambient authority token on to it. This token is not forgeable since the `AmbientAuth` constructor is private—only the runtime glue code
```

Reply:
I think I solved most if not all of this now.

Regarding the use of "token", it's mentioned again right below the quote, in `which holds the unforgeable `AmbientAuth` token in its root field`. Is that enough to guide the reader?